Intro to coding - Improved (#2716)
enkidevs/curriculum: coding-intro/coding-intro-core/core-concepts/think-like-a-coder.md

---
author: abdullah

tags:
  - discussion

type: normal

> category: discussion

Reviewer: here too, making this insight a `discussion` will display the "Give it a try" button, but there is nothing to try here

Author: Trying to make this a discussion insight as well.

---

> 

Reviewer: ```suggestion
```

> # Think like a coder

Reviewer: let's capitalise the title

```suggestion
# Think Like a Coder
```

---
## Content

Coding is all about saving time (and sometimes about making money 😉).

Let's look at a scenario.

> Imagine you have to simulate grabbing 5 coins from 3 different jars. 

Reviewer: ```suggestion
Imagine you have to simulate grabbing five coins from three different jars. 
```

> You could make a function, `pick_coins`, with 3 commands and use it 5 times, saving yourself a lot of time.

Reviewer: ```suggestion
You could make a function, `pick_coins`, with three commands and use it five times, saving yourself a lot of time.
```

> ```python

Reviewer: you set the language tag to `python` but you're using curly braces when defining functions

```suggestion
```js
```

> pick_coins(){
> #Grab coin 1
> #Grab coin 2
> #Grab coin 3
> }
> 
> pick_coins()
> pick_coins()
> pick_coins()
> pick_coins()
> pick_coins()

Reviewer: ```suggestion
pick_coins() {
  # Grab the first coin
  # Grab the second coin
  # Grab the third coin
}

pick_coins()
pick_coins()
pick_coins()
pick_coins()
pick_coins()
```
